Add Password for Garnet

Alirexaa opened the pull request:
## Description

The implementation is based on https://microsoft.github.io/garnet/docs/security#password-protected-sessions.
Also, I override the `entrypoint` from `/app/GarnetServer` to `/bin/sh` to provide the password from environment variables.

Contributes to #6155

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - x ] Yes
    - If yes, did you have an API Review for it?
      - [x] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [x] Yes
      - [ ] No
  - [ ] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [ ] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Is this introducing a breaking change?
      - [ ] Yes
        - Link to aspire-docs issue (please use this [`breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)):
      - [ ] No
        - Link to aspire-docs issue (please use this [`doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)):
  - [ ] No

Alirexaa committed:
Add Password for Garnet

Alirexaa committed:
Update tests

Alirexaa committed:
Update playground manifest

davidfowl commented:
Lets make sure we get the redis implementation working and merge end to end (including deployment to azure with azd) before we review this one. Unlike redis, we can influence garnet to add an env variable instead of using arguments 😄 for passwords.

cc @ReubenBond

eerhardt committed:
Merge remote-tracking branch 'upstream/main' into garnet-password

eerhardt committed:
Fix up merge

eerhardt committed:
PR feedback

1. Don't use special characters until azd gets fixed
2. Refactor code to be simplified

eerhardt reviewed (approved):
Thanks again for the great work @Alirexaa. I merged main and pushed some minor feedback - mostly to make it consistent with updates to the Redis implementation.

I think this is ready to merge.
